File: python-api/src/api/handlers/polls/getPoll/getPollHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['POLLS_TABLE_NAME']
DYNAMO_POLLS_ID = os.environ['POLLS_ID']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def getPoll(event, context):

    pk = event['pathParameters']['id']
    print('pk: ', pk)

    pk = DYNAMO_POLLS_ID + '#' + pk
    params = {
        'Key': {
            'PK': pk,
            'SK': pk
        }
    }

    try:
        table = dynamodb_client.Table(TABLE_NAME)
        response = table.get_item(**params)
    except Exception as ex:
        return create_response(500, str(ex))

    print('response: ', response)

    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
      return create_response(500, 'Error getting the Poll')

    if not 'Item' in response:
        return create_response(404, 'Poll with id: ' + pk + ' not found')

    poll = response['Item']

    return create_response(200, poll)

File: nodeJS/src/api/handlers/polls/getPoll/getPollHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})

const tableName = process.env.POLLS_TABLE_NAME
const dynamoPollsId = process.env.POLLS_ID

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const getPoll = async (event) => {
    const PK = event.pathParameters.id
    console.log('PK: ', PK)
    
    const params = {
      TableName: tableName,
      Key: {
        PK: dynamoPollsId + '#' + PK,
        SK: dynamoPollsId + '#' + PK,
      }
    }
  
    try {
      const pollSearch = await db.get(params).promise()
      console.log('poll: ', pollSearch)
  
      if (!pollSearch.Item) {
        return response(404, 'Poll with id: ' + PK + 'not found')
      }
  
      const poll = pollSearch.Item
      return response(200, poll)
    } catch (err) {
      console.log('err: ', err)
      return response(500, 'Error getting the Poll')
    }
}

module.exports = {
    getPoll
}